### PR TITLE
api-doc: API doc for thirdpartyrepo endpoints.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
In the context of the swaggo lib implement the api doc for thirdpartyrepo endpoints.
FIXES:
- https://issues.redhat.com/browse/THEEDGE-3392
- https://issues.redhat.com/browse/THEEDGE-3346
- https://issues.redhat.com/browse/THEEDGE-3385
- https://issues.redhat.com/browse/THEEDGE-3397

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by de1987 — APPROVED** [review]

**Review by acosferreira — APPROVED** [review]
lgtm

File: pkg/routes/thirdpartyrepo.go [hunk 1]
```diff
@@ -59,16 +59,16 @@ var thirdPartyRepoFilters = common.ComposeFilters(
...
-// @Param		 required_parm query string true "A placeholder for required parameter" example(cat)
-// @Param		 optional_parm query int false "A placeholder for optional parameter" example(42)
-// @Success      200 {object} models.SuccessPlaceholderResponse
-// @Failure      400 {object} errors.BadRequest
-// @Failure      500 {object} errors.InternalServerError
+// @Param		 name	path	string	true	"ThirdParty repo Name" example(my_custom_repo)
+// @Success      200 {object} models.CheckThirdPartyRepoNameAPI "The third party repository name check result"
+// @Failure      400 {object} errors.BadRequest "The request sent couldn't be processed."
+// @Failure      500 {object} errors.InternalServerError "There was an internal server error."
 // @Router       /thirdpartyrepo/{name} [get]
```
Comment (mgold1234) [inline pkg/routes/thirdpartyrepo.go:72]: I see that this is the end point

File: pkg/routes/thirdpartyrepo.go [hunk 2]
```diff
@@ -59,16 +59,16 @@ var thirdPartyRepoFilters = common.ComposeFilters(
...
+// @Description  Checks to see if a ThirdParty repo Name exists.
 // @Tags         Third Party Repo
 // @Accept       json
 // @Produce      json
-// @Param		 required_parm query string true "A placeholder for required parameter" example(cat)
-// @Param		 optional_parm query int false "A placeholder for optional parameter" example(42)
-// @Success      200 {object} models.SuccessPlaceholderResponse
-// @Failure      400 {object} errors.BadRequest
-// @Failure      500 {object} errors.InternalServerError
+// @Param		 name	path	string	true	"ThirdParty repo Name" example(my_custom_repo)
```
Comment (mgold1234) [inline pkg/routes/thirdpartyrepo.go:68]: align

File: pkg/routes/thirdpartyrepo.go [hunk 3]
```diff
@@ -311,16 +312,21 @@ func GetAllContentSourcesRepositories(w http.ResponseWriter, r *http.Request) {
...
 // @Tags         Third Party Repo
 // @Accept       json
 // @Produce      json
-// @Param		 required_parm query string true "A placeholder for required parameter" example(cat)
-// @Param		 optional_parm query int false "A placeholder for optional parameter" example(42)
-// @Success      200 {object} models.SuccessPlaceholderResponse
-// @Failure      400 {object} errors.BadRequest
-// @Failure      500 {object} errors.InternalServerError
+// @Param		 sort_by query string false "fields: created_at, name, updated_at. To sort DESC use - before the fields." example(-name)
+// @Param		 name query string false "field: filter by name" example(my_repo)
```
Comment (mgold1234) [inline pkg/routes/thirdpartyrepo.go:322]: align

File: pkg/routes/thirdpartyrepo.go [hunk 4]
```diff
@@ -435,17 +442,18 @@ func GetThirdPartyRepoByID(w http.ResponseWriter, r *http.Request) {
...
 // @Tags         Third Party Repo
 // @Accept       json
 // @Produce      json
-// @Param		 required_parm path string true "A placeholder for required parameter" example(cat)
-// @Param		 optional_parm path int false "A placeholder for optional parameter" example(42)
-// @Param        body	body	models.Image	true	"request body"
-// @Success      200 {object} models.SuccessPlaceholderResponse
-// @Failure      400 {object} errors.BadRequest
-// @Failure      500 {object} errors.InternalServerError
+// @Param		 ID query int true "An unique existing third party repository id." example(1080)
```
Comment (mgold1234) [inline pkg/routes/thirdpartyrepo.go:451]: align

File: pkg/routes/thirdpartyrepo.go [hunk 5]
```diff
@@ -491,15 +499,17 @@ func UpdateThirdPartyRepo(w http.ResponseWriter, r *http.Request) {
...
+// @ID           DeleteThirdPartyRepoByID
+// @Description  Delete third party repository using id.
 // @Tags         Third Party Repo
 // @Accept       json
 // @Produce      json
-// @Param		 imageSetID		path    int  true  "Identifier of the ImageSet"
-// @Success      200 {object} models.SuccessPlaceholderResponse
-// @Failure      400 {object} errors.BadRequest
-// @Failure      500 {object} errors.InternalServerError
+// @Param		 ID query int true "An unique existing third party repository id." example(1080)
```
Comment (mgold1234) [inline pkg/routes/thirdpartyrepo.go:508]: align

File: pkg/routes/thirdpartyrepo.go (hunk 1 above)
Comment (ldjebran) [reply pkg/routes/thirdpartyrepo.go:72]: fixed in PR https://github.com/RedHatInsights/edge-api/pull/2154

File: pkg/routes/thirdpartyrepo.go (hunk 3 above)
Comment (ldjebran) [reply pkg/routes/thirdpartyrepo.go:322]: fixed in PR https://github.com/RedHatInsights/edge-api/pull/2154

File: pkg/routes/thirdpartyrepo.go (hunk 4 above)
Comment (ldjebran) [reply pkg/routes/thirdpartyrepo.go:451]: fixed in PR https://github.com/RedHatInsights/edge-api/pull/2154

File: pkg/routes/thirdpartyrepo.go (hunk 5 above)
Comment (ldjebran) [reply pkg/routes/thirdpartyrepo.go:508]: fixed in PR https://github.com/RedHatInsights/edge-api/pull/2154

File: pkg/routes/thirdpartyrepo.go (hunk 2 above)
Comment (ldjebran) [reply pkg/routes/thirdpartyrepo.go:68]: fixed in PR https://github.com/RedHatInsights/edge-api/pull/2154

**Pull request merged**